File: Python/71. Simplify Path.py
class Solution:
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        tmp = path.split('/')
        ans = list()
        for name in tmp:
            if name == '..':
                if len(ans) > 0:
                    ans.pop()
            elif name == '.':
                continue
            elif name != '':
                ans.append(name)
        if len(ans) == 0:
            return '/'
        ans = '/'.join(ans)
        return '/' + ans
                

File: C++/71. Simplify Path.cpp
class Solution {
public:
    string simplifyPath(string path) {
        vector<string> res;
        string tmp, ans ;
        int i = 0, len = path.length();
        while(i<len)
        {
            string tmp = "";
            while(path[i] != '/' && i<len)
            {
                tmp += path[i];
                i++;
            }
            i++;
            if(tmp == ".")
                continue;
            else if(tmp == "..")
            {
                if(res.size() > 0)
                    res.pop_back();
            }
            else if(tmp.length() > 0)
                res.push_back(tmp);
        }
        if(res.size() == 0)
            return "/";
        for(auto s: res)
            ans = ans + "/" + s;
        return ans;
            
    }
};